internal/manifest/statecheck/state_check.go

```diff
@@ -0,0 +1,88 @@
+package statecheck
+
+import (
+	"context"
+
+	apiappsv1 "k8s.io/api/apps/v1"
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+)
+
+type ManagerStateCheck struct {
+	StatefulSetChecker     StatefulSetStateChecker
+	DeploymentStateChecker DeploymentStateChecker
+}
+
+type DeploymentStateChecker interface {
+	GetState(deploy *apiappsv1.Deployment) (shared.State, error)
+}
+
+type StatefulSetStateChecker interface {
+	GetState(ctx context.Context, clnt client.Client, statefulSet *apiappsv1.StatefulSet) (shared.State, error)
+}
+
+type ManagerKind string
+
+const (
+	DeploymentKind  ManagerKind = "Deployment"
+	StatefulSetKind ManagerKind = "StatefulSet"
+)
+
+type Manager struct {
+	Kind ManagerKind
+	*apiappsv1.Deployment
+	*apiappsv1.StatefulSet
+}
+
+func NewManagerStateCheck(statefulSetChecker StatefulSetStateChecker,
+	deploymentChecker DeploymentStateChecker,
+) *ManagerStateCheck {
+	return &ManagerStateCheck{
+		StatefulSetChecker:     statefulSetChecker,
+		DeploymentStateChecker: deploymentChecker,
+	}
+}
+
+func (m *ManagerStateCheck) GetState(ctx context.Context,
+	clnt client.Client,
+	resources []*resource.Info,
+) (shared.State, error) {
+	mgr := findManager(clnt, resources)
+	if mgr == nil {
+		return shared.StateReady, nil
+	}
```

Comment:
Just wondering, no need to address in this PR... Do you think it is reasonable that we are defaulting to Ready if we can't determine the state? My understanding is that we always expect a manager to be present. Maybe it would be a good idea to introduce another state `Unknown` that we set when we are not able to determine the state correctly. Any opinions? If you agree that we may need to do something about it I can create a separate ticket and discuss it with the team.

Reply:
this was a temporary solution until the StatefulSet was introduced. But since it is now already introduced, we can even remove this

Reply:
Let's discuss with the team what would be the reasonable thing here 👍🏻